### PR TITLE
More complete tests against gennodejs

### PR DESCRIPTION
**Pull request by chris-smith**
Requires test_msgs package available at https://github.com/RethinkRobotics-opensource/test_msgs.

### CODE REVIEW

File: test/genjsTest.js [hunk 1]
```diff
@@ -281,4 +286,308 @@ describe('genjsTests', () => {
...
+      expect(deserializedRequest2.header_response.frame_id).to.equal(frameId);
+      expect(deserializedRequest2.header_response.stamp.secs).to.equal(0);
+      expect(deserializedRequest2.header_response.stamp.nsecs).to.equal(0);
+
+      done();
+    });
+  });
+
+  describe('actions', () => {
+    // TODO: TEST actions
```
Comment (IanTheEngineer) [inline test/genjsTest.js:591]: We should track this in an issue, so as not to forget.

**Comment by IanTheEngineer** [comment]
+1 looks good to me. I'm going to merge as-is. Should we similarly test against ROS Services and a `test-srvs` package?

**Comment by chris-smith** [comment]
`test_msgs` actually includes service and action definitions as well.

File: test/genjsTest.js [hunk 2]
```diff
@@ -281,4 +286,308 @@ describe('genjsTests', () => {
...
+      expect(deserializedMsg2.array_field.length).to.equal(arrLen2);
+      deserializedMsg2.array_field.forEach((item) => {
+        expect(item).to.be.an.instanceof(BaseType);
+      });
+
+      done();
+    });
+
+    it('services and constants', (done) => {
+      const BasicService = msgUtils.getHandlerForSrvType('test_msgs/BasicService');
```
Comment (IanTheEngineer) [inline test/genjsTest.js:430]: I kinda glossed over that one. Good call. You've got a solid set of tests here

**Pull request merged**